packages/target-nodejs-sdk/test/telemetry.spec.js

```diff
@@ -0,0 +1,422 @@
+          }
+        ]
+      },
+      {
+        index: 2,
+        name: "remote-only-mbox-a",
+        options: [
+          {
+            content: {
+              paymentExperience: "alpha10",
+              showFeatureX: true,
+              paymentGatewayVersion: 3.1,
+              customerFeedbackValue: 99
+            },
+            type: "json",
+            eventToken:
+              "8MDIALF7bsTPYn79fLBNQpNWHtnQtQrJfmRrQugEa2qCnQ9Y9OaLL2gsdrWQTvE54PwSz67rmXWmSnkXpSSS2Q=="
+          }
+        ]
+      }
+    ]
+  }
+};
+
+const context = {
+  channel: "web",
+  mobilePlatform: null,
+  application: null,
+  screen: null,
+  window: null,
+  browser: null,
+  address: {
+    url: "http://adobe.com",
+    referringUrl: null
+  },
+  geo: null,
+  timeOffsetInMinutes: null,
+  userAgent:
+    "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:73.0) Gecko/20100101 Firefox/73.0",
+  beacon: false
+};
+
+const visitorId = {
+  tntId: "338e3c1e51f7416a8e1ccba4f81acea0.28_0",
+  marketingCloudVisitorId: "07327024324407615852294135870030620007"
+};
+
+const targetRequest = {
+  id: visitorId,
+  context
+};
+
+const targetClientOptions = {
+  client: "someClientId",
+  organizationId: "someOrgId",
+  artifactFormat: "json", // setting this tells the artifactProvider deobfuscation is not needed
+  targetLocationHint: "28",
+  pollingInterval: 0,
+  maximumWaitReady: 500,
+  telemetryEnabled: true
+};
+
+describe("telemetry mode", () => {
+  let client;
+
+  beforeEach(() => {
+    fetch.resetMocks();
+    if (client) {
+      client = undefined;
+    }
+  });
+
+  describe("on-device", () => {
+    it("immediately executes telemetries", () => {
+      expect.assertions(1);
+
+      return new Promise(done => {
+        fetch
+          .once(JSON.stringify(DECISIONING_PAYLOAD_FEATURE_FLAG))
+          .once(JSON.stringify(DELIVERY_RESPONSE));
+
+        async function clientReady() {
+          await client.getOffers({
+            request: {
+              ...targetRequest,
+              prefetch: {
+                mboxes: [
+                  {
+                    name: "mbox-feature-flags",
+                    index: 1
+                  },
+                  {
+                    name: "remote-only-mbox-a",
+                    index: 2
+                  },
+                  {
+                    name: "remote-only-mbox-b",
+                    index: 2
+                  }
+                ]
+              }
+            },
+            sessionId: "dummy_session"
+          });
+
+          setTimeout(() => {
+            expect(executeTelemetries.mock.calls.length).toBe(1);
+            done();
+          }, 500);
+        }
+
+        client = TargetClient.create({
+          decisioningMethod: DECISIONING_METHOD.ON_DEVICE,
+          ...targetClientOptions,
+          events: { clientReady }
+        });
+      });
+    });
+  });
+
+  describe("hybrid", () => {
+    it("calls sendNotifications when on-device used", () => {
```

Comment:
Let's rename this to `executes telemetries as part of sendNotifications call when on-device used` so it's a bit more clear